src/topology/bases.lean

```diff
@@ -9,10 +9,40 @@ import topology.continuous_on
 /-!
 # Bases of topologies. Countability axioms.
 
+A topological basis on a topological space `t` is a collection of sets,
+such that taking the union of these sets generates all open sets, without the need to take
+finite intersections of them. This file introduces a framework for dealing with these collections,
+and also what more we can say under certain countability conditions on bases,
+which are referred to as first- and second-countable.
+We also briefly cover the theory of separable spaces, which are those with a countable, dense
+subset. If a space is second-countable, and also has a countably generated uniformity filter
+(for example, if `t` is a metric space), it will automatically be separable (and indeed, these
+conditions are equivalent in this case).
+
+## Main definitions
+
+* `is_topological_basis s`: Indicates whether the topology `t` is generated by `s`, but only
+  allowing unions of sets in `s`.
+* `separable_space α`: Indicates whether there is a countable, dense subset of `t`.
+* `first_countable_topology α`: A topology which, at any point `x`, `𝓝 x` is countably generated.
+* `second_countable_topology α`: A topology which has a topological basis which is countable.
+
+## Main results
+
+* `first_countable_topology.tendsto_subseq`: In a first-countable space,
+  cluster points are limits of subsequences.
+* `second_countable_topology.is_open_Union_countable`: In a second-countable space, the union of
+  arbitrarily-many open sets is equal to a sub-union of only countably many of these sets.
+* `second_countable_topology.countable_cover_nhds`: Consider a function `f` with the property that
+  `f x ∈ 𝓝 x` for all `x`. Then there is some countable set `s` whose image covers the space.
```

Comment:
I'd include the signature of `f` here, so `f x ∈ 𝓝 x` doesn't come as a surprise.